packages/protocol/engine/notarization/manager.go

```diff
@@ -45,7 +52,13 @@ func NewManager(storage *storage.Storage, sybilProtection *sybilprotection.Sybil
 		pendingConflictsCounters:   shrinkingmap.New[epoch.Index, uint64](),
 		acceptanceTime:             storage.Settings.LatestCommitment().Index().EndTime(),
 		optsMinCommittableEpochAge: defaultMinEpochCommittableAge,
-	}, opts)
+
+		evictionState: evictionState,
+	}, opts, (*Manager).setupEvents)
+}
+
+func (m *Manager) setupEvents() {
+	m.evictionState.Events.EpochEvicted.Hook(event.NewClosure(m.evictUntil))
```

Comment:
This doesn't seem right. Also if it's like this we probably call `evictUntil` twice: here and in `(m *EpochMutations) Commit`. I think the cleanest way would be to add this information to `Events.EpochCommitted` as we get all the information in `createCommitment` in line `acceptedBlocks, acceptedTransactions, activeValidators, cumulativeWeight, err := m.EpochMutations.Commit(index)`. Then you can also get rid of the getters in `EpochMutations`

Reply:
ok, extending EpochCommitted looks kinda nicer, and actually, the `acceptedBlocks`, and the rest still exist due to reference, which I took wrongly the last time